src/main/proto/ga4gh/read_service.proto

```diff
@@ -91,6 +99,49 @@ message GetReadGroupSetRequest {
   string read_group_set_id = 1;
 }
 
+// ******************  /readgroups  *********************/
+
+// This request maps to the body of `POST /readgroups/search` as JSON.
+
+
+message SearchReadGroupsRequest {
+  // The dataset to search (required).
+  string dataset_id = 1;
```

Comment:
> It would simplified the API if we didn't push dataset down to
the lower levels unless there is a compelling use case and stick
to the container hierarchy.  The container hierarchy will serve
the `find all' type of queries (walking the hierarchy) and
criteria queries can build in this plus the metadata.

I completely agree, but I don't think we have a container hierarchy officially. Attempts to define the formal data model as a hierarchy of containers have failed because of an insistence on the importance of multiple occupancy (e.g., references belong to multiple reference sets, callsets belong to multiple variants sets, etc). Given that servers can make shallow copies of objects I have never understood the necessity of this. Until we actually understand what being "in" a dataset means, we're not going to solve these problems.